File: src/main/java/ru/trukhmanov/twochairsbackend/util/CurrentUser.java
package ru.trukhmanov.twochairsbackend.util;

import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;

public final class CurrentUser{
    private CurrentUser(){}

    public static long id(){
        Authentication a = SecurityContextHolder.getContext().getAuthentication();
        if(a == null || a.getPrincipal() == null) throw new IllegalStateException("Unauthenticated");
        return Long.parseLong(a.getPrincipal().toString());
    }
}

File: src/main/kotlin/ru/trukhmanov/twochairsbackend/util/CurrentUser.kt
package ru.trukhmanov.twochairsbackend.util

import org.springframework.security.core.context.SecurityContextHolder

object CurrentUser {
    fun id(): Long {
        val authentication = SecurityContextHolder.getContext().authentication
        val principal = authentication?.principal ?: throw IllegalStateException("Unauthenticated")
        return principal.toString().toLong()
    }
}
